js/data/array_utils.js

```diff
@@ -96,16 +96,16 @@ function applyBatch({ keyInfo, data, changes, groupCount, useInsertIndex, immuta
         !disableCache && generateDataByKeyMap(keyInfo, items);
 
         switch(item.type) {
-            case 'update': update(keyInfo, items, item.key, item.data, true, immutable); break;
-            case 'insert': insert(keyInfo, items, item.data, useInsertIndex && isDefined(item.index) ? item.index : -1, true); break;
-            case 'remove': remove(keyInfo, items, item.key, true); break;
+            case 'update': update(keyInfo, items, item.key, item.data, true, immutable, logError); break;
+            case 'insert': insert(keyInfo, items, item.data, useInsertIndex && isDefined(item.index) ? item.index : -1, true, logError); break;
+            case 'remove': remove(keyInfo, items, item.key, true, logError); break;
         }
     });
     return resultItems;
 }
 
-function getErrorResult(isBatch, errorCode) {
-    return !isBatch ? rejectedPromise(errors.Error(errorCode)) : errors.log(errorCode);
+function getErrorResult(isBatch, logError, errorCode) {
+    return !isBatch ? rejectedPromise(errors.Error(errorCode)) : logError === true && errors.log(errorCode);
```

Comment:
why `logError === true` and not simple `logError`?

Reply:
Will change to logError:)